storm-client/src/jvm/org/apache/storm/blobstore/BlobStore.java

```diff
@@ -288,13 +288,15 @@ public void createBlob(String key, InputStream in, SettableBlobMeta meta, Subjec
             while ((len = in.read(buffer)) > 0) {
                 out.write(buffer, 0, len);
             }
-            out.close();
-        } catch (AuthorizationException | IOException | RuntimeException e) {
```

Comment:
The documentation for this method indicates that we throw the above exceptions, however they were being caught instead of thrown. This change will result in the exceptions being thrown while still helping to guarantee the `out` stream is closed as the `in` stream was already being ensured to be closed.

Reply:
It looks to me that the exception is still caught instead of thrown?

Reply:
Formerly they were caught. Removing these lines will result in them being thrown as both the method documentation and declarations intend.

We may not want this change however, and instead we could update the declaration and the documentation such that the behavior remains unchanged. I am willing to do either one, as this PR is focused on eliminating the possibility of leaking some sockets.